ko/pl.rst

```diff
@@ -0,0 +1,1383 @@
+| LOGIN_DENIED        | 유효하지 않는 사용자 이름이나 암호로 DBMS에 로그인 시도          |
++---------------------+------------------------------------------------------------------+
+| NO_DATA_FOUND       | SELECT INTO 문 실행 결과 0개의 Row가 반환됨                      |
++---------------------+------------------------------------------------------------------+
+| PROGRAM_ERROR       | 시스템 내부 에러                                                 |
++---------------------+------------------------------------------------------------------+
+| ROWTYPE_MISMATCH    | 조회 결과와 결과를 담을 호스트 변수 사이에 타입이 호환되지 않음  |
++---------------------+------------------------------------------------------------------+
+| STORAGE_ERROR       | 메모리 부족으로 인한 할당 실패                                   |
++---------------------+------------------------------------------------------------------+
+| TOO_MANY_ROWS       | SELECT INTO 문 실행 결과 2개 이상의 Row가 반환됨                 |
++---------------------+------------------------------------------------------------------+
+| VALUE_ERROR         | 명시적/암묵적 형변환 실패                                        |
++---------------------+------------------------------------------------------------------+
+| ZERO_DIVIDE         | 0으로 나누기 시도                                                |
++---------------------+------------------------------------------------------------------+
+
+.. code-block:: sql
+
+    CREATE FUNCTION athlete_code(name VARCHAR) RETURN integer AS
+        c INTEGER;
+    BEGIN
+        -- SELECT INTO 문은 단 하나, 그리고 오직 하나의 Row를 결과로 가져야 함
+        SELECT code INTO c FROM athlete a WHERE a.name = name;
+        RETURN c;
+    EXCEPTION
+        WHEN NO_DATA_FOUND THEN
+            PUT_LINE('error: no rows found for athlete name ' || name);
+            RETURN -1;
+        WHEN TOO_MANY_ROWS THEN
+            PUT_LINE('error: more than one rows found for athlete name ' || name);
+            RETURN -1;
+    END;
+
+.. _decl:
+
+******************
+선언문
+******************
+
+프로시저나 함수 선언문, 그리고 Block 실행문에는 선언부 *seq_of_declare_specs*\가 존재한다.
+선언부에서는 아래 문법에서 정의하는 바와 같이 변수, 상수, Exception, 커서,
+내부 프로시저/함수, Autonomous Transaction 여부를 선언할 수 있다.
+선언된 각 항목들은 해당 선언부를 뒤따르는 *body* 안에서 참조할 수 있다.
+::
+
+    <seq_of_declare_specs> ::= <declare_spec> [ <declare_spec> ... ]
+    <declare_spec> ::=
+          <variable_decl>
+        | <constant_decl>
+        | <exception_decl>
+        | <cursor_decl>
+        | <inner_procedure_decl>
+        | <inner_function_decl>
+        | <autonomous_transaction_decl>
+
+선언 가능한 각 항목에 대한 설명은 아래 내용을 참고한다.
+
+변수 선언
+=========
+::
+
+    <variable_decl> ::=
+        <identifier> <type_spec> [ [ NOT NULL ] <initial_value_part> ] ;
+
+    <initial_value_part> ::= { := | DEFAULT } <expression>
+
+* *type_spec*: :ref:`데이터 타입 <types>` 절에서 설명한 시스템 제공 타입
+
+변수 선언에 선택적으로 NOT NULL 조건과 초기값을 지정할 수 있다.
+NOT NULL 조건이 지정된 경우 반드시 NULL이 아닌 초기값이 함께 지정되어야 한다.
+선언할 때 초기값이 지정되지 않은 변수는 암묵적으로 NULL 값을 갖게 된다.
+
+.. code-block:: sql
+
+    CREATE PROCEDURE test_proc AS
```

Comment:
Don't we intentionally use DECLARE at the top?

Reply:
No need to put DECLARE in this case, and the syntax forbids it.
DECLARE is used in the Block statement.